packages/venia-concept/src/index.js

```diff
@@ -39,14 +40,38 @@ const authLink = setContext((_, { headers }) => {
 });
 
 // https://www.apollographql.com/docs/link/links/error/
-const errorLink = onError(({ graphQLErrors, networkError }) => {
+const errorLink = onError(({ graphQLErrors, networkError, response }) => {
     if (graphQLErrors)
         graphQLErrors.forEach(({ message, locations, path }) =>
             console.log(
                 `[GraphQL error]: Message: ${message}, Location: ${locations}, Path: ${path}`
             )
         );
     if (networkError) console.log(`[Network error]: ${networkError}`);
+
+    const { data, errors } = response;
+
+    errors.forEach(({ message, path }, index) => {
+        // It's within the GraphQL spec to receive data and errors, where errors are merely informational and not
+        // intended to block. Almost all existing components were not built with this in mind, so we build special
+        // handling of this error message so we can deal with it at the time we deem appropriate.
+        if (message === 'Some of the products are out of stock.') {
+            const pathToCartItems = path.slice(0, -1);
+            const cartItems = getWithPath(data, pathToCartItems);
+
+            // Until MC-36092 is resolved, we need to guard against a null product being returned as well.
+            if (cartItems[0] === null) {
```

Comment:
Is it always the first item or is it possible it could just exist somewhere in the array? May be more future proof to do this in a loop, at least until the noted fix is complete.

Reply:
Regardless of the position of the OOS item, or even if there are multiple OOS items, `null` always shows up at the first index. Whatever the bug is, at least it's consistent.

A loop seems overkill, but I'm happy to use the index sliced out of the path that seems to reference the `null` item. GraphQL **is** fixing this in 2.4.1, so future proofing shouldn't really be necessary.

Reply:
If there are multiple OOS items are there multiple `null` products, or is there always just one?

Reply:
Correct, always just one.